### DISCUSSION TITLE
Window size

### DISCUSSION THREAD

**Discussion by garygeeg** [DISTRHO/Cardinal#91] [Q&A]
Hi,

Trying out Cardinal in Reaper and Ableton, running really well overall, good to have an alternative to the commercial version.

One little thing:

Should I be able to resize the plugins window? Seems set at a certain size, not unworkable at all but the ability to have it nearer full screen would be great. This is the VST2 version, VST3 just causes Reaper to disappear when I load it.

Cheers.

**Comment by falkTX** [comment]
VST3 is experimental, the VST2 should be resizable yes.
The bottom-right ought to have a resize handle.

Which OS are you using? and version?

**Reply by garygeeg** [reply]
Windows 10, 64 bit Reaper 6.37

I just tried the release from earlier today to ensure I had the latest version (not sure where the version is listed? f998ee2?) but don't see any resize handle, neither instrument or synth version. On a positive note, the duplicate keypress issue is fixed in Reaper, thanks!

**Reply by falkTX** [reply]
Does the plugin window perhaps go above the bounds of the window created by reaper?
I have seen it happening in macOS due to me having what amounts to ~4x scale factor (it is on a tv) so because of the size I end up not seeing the resize handle.

If you post a screenshot I know what to look for, basically the horizontal and vertical sliders need to be in frame.

**Reply by subderground** [reply]
Here with Reaper (last update + Windows 11 64bit), I can resize perfectly the windows of Cardinal VST3 22.05 while using the horizontally or vertically handle. But if I try to modify the size with the bottom-right one, that doesn't work (black screen with multiple supervision). It's not a big bug.
(edit)  With VST2, the horizontally/vertically handles don't work while the bottom-right one is fine ;-) 

In VST3 version, another one is the impossibility to use "back space" and "delete" keys in the text zones (search module zone or in CF Label plugin" 
_(Also, an equivalent plugin to ''Glue'', to add labels everywhere on the GUI, should be a must have ;-) )._
(edit) With VST2 version, "back space", "delete" and "space keys work well, but it's impossible use "space bar" to "start/stop" the daw while Cardinal gui is open.

**Comment by garygeeg** [comment]
![Screenshot 2021-12-12 231708](https://user-images.githubusercontent.com/95886416/145733661-8578f8a9-f309-4b09-b4ba-2e030606dee3.png)

I get the regular Reaper diagonal arrow cursor when I hover over the bottom right and can drag Reapers window bigger (the white border, right and bottom in the above).

I'm on a laptop normally with windows scaling at 125% so I put that back to 100% in case that was affecting anything. The scrollbars seem to perform correctly, I don't lose part of them when I scroll to extremes etc. I checked in Live too and don't see anything there either.

Thanks.

**Comment by falkTX** [comment]
The resize handle should be visible in the upcoming build https://github.com/DISTRHO/Cardinal/actions/runs/1580151380

**Reply by garygeeg** [reply]
Thanks, I can now see the resize handle but unfortunately it doesn't appear to do anything. The cursor doesn't change when I hover over it, certainly nothing happens if I click drag it. Tested in reaper and ableton, also the 32bit version in Reaper but all exactly the same for me.

I assume this works for others, is there anything I can look at on my laptop to help troubleshoot? I thought it was fairly normal, latest gen i5, Intel XE graphics, windows 10 etc.

Thanks for your efforts with this.

**Reply by falkTX** [reply]
The cursor not changing is what I am looking at next.
I have a way to continue the work on this for the time being, so nothing that needs testing at the moment.
Will try to confirm the non-resizable issue.

Once I am done with the changes I will reply here, which then finally needs testing. Just let me try a couple of things first.

**Reply by garygeeg** [reply]
Oh ok, I thought you meant the resizing should work now, apologies.

Thanks.

**Reply by falkTX** [reply]
I thought it did, but you reported it not to, so I have to investigate now.
Might take a bit of time.

**Reply by zoikkis** [reply]
For me the handle works under Linux but not under Windows 10 (Reaper and Renoise). The handle is there but doesn't react to left click + drag

**Comment by RO-mix** [comment]
LV2 version in Reaper on Win64 can be resized. VST version - not.

**Comment by subderground** [comment]
Hi, 
V 22.12 :  VST3 is fine but LV2 version has the "resizing glitch".
Btw, it's not a big problem.
See picture below :-)
![gui-resizing-glitches-LV2](https://user-images.githubusercontent.com/105162700/208914691-84cb226a-6c83-4761-85dc-f84f9a423a53.png)

**Reply by falkTX** [reply]
that is a host issue. Cardinal and all DPF based plugins do not use extensions to do UI resizing.
the host is supposed to catch the OS resize event and adjust its window accordingly.
at the time of writing, only Carla does this properly.
